tests/serverpod_test_server/test/core_middleware_ordering_test.dart

```diff
@@ -0,0 +1,262 @@
+import 'package:http/http.dart' as http;
+import 'package:serverpod/serverpod.dart';
+import 'package:test/test.dart';
+
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:serverpod_test_server/src/generated/endpoints.dart';
+
+/// Tests to verify that user middleware cannot bypass core middleware
+/// (_headers and _reportException).
```

Comment:
I'm still wondering if we might want to allow overriding all middleware. But fine for now..